robot/control/Src/modules/LEDModule.cpp

```diff
@@ -168,4 +198,52 @@ void LEDModule::setColor(uint32_t led0, uint32_t led1) {
     dotStarNCS.write(false);
     dotStarSPI.lock()->transmit(data);
     dotStarNCS.write(true);
-}
\ No newline at end of file
+}
+
+void LEDModule::displayErrors() {
+    // Check if there are errors to display, else exit
+    if (colorQueue.size() == 0) {
+        return;
+    }
+
+    if (lightsOn && framesOnCounter >= framesOn){
+        // If error lights have been on long enough, switch them off
+        framesOnCounter = 0;
+        lightsOn = false;
+        setColor(0x000000,0x000000, 0x000000);
+    } else if (!lightsOn && framesOffCounter >= framesOff) {
+        // If error lights have been off long enough, switch them back on to next error color
+        framesOffCounter = 0;
+        lightsOn = true;
+
+        // Make sure index loops back if it has reached end of vector
+        if (++index > colorQueue.size()-1) {
+            index = 0;
+        }
```

Comment:
This took me a second to figure out, I usually see it as `index = (index + 1) % colorQueue.size();`

Reply:
That's actually a very concise way to write it, since modular math always wraps things back around. I like it!